tests/conftest.py

```diff
@@ -7128,16 +7128,21 @@ def factory(
         return primary_cluster_instances, secondary_cluster_instances
 
     def teardown():
-        acm_obj = AcmAddClusters()
+        failed_to_delete = []
         managed_cluster_instances = [
             primary_cluster_instances,
             secondary_cluster_instances,
         ]
-        app_list = []
         for cluster_instances in managed_cluster_instances:
             for apps in cluster_instances:
-                app_list.append(apps.app_name)
-        dr_helpers_ui.delete_application_ui(acm_obj, workloads_to_delete=app_list)
+                try:
+                    apps.delete_workload()
```

Comment:
@avd-sagare This implementation may not work/clean up properly as we are creating resources with unique resource names for almost everything eg: channel, namespaces, pods, DRPC, PVCs and stuff. The delete_workload() might need some changes according to the new implementation. Please check it deletes everything at-least for one combo when apps deployed on (Primary=True, Secondary_false) and for the rest as we decided let's create the ocs-ci issue and we can move ahead to merge this PR. 

I am afraid we will have leftovers as we are creating unique_resources to cut out the duplication when multiple apps are deployed on both the managed clusters

Please attach the teardown logs for this PR validation.

Reply:
@Shrivaibavi I have checked about resources like pods, pvc,vrg,drpc got deleted after test. Please have a look and let me know if anything is missing.  
https://jenkins-csb-odf-qe-ocs4.dno.corp.redhat.com/view/DR/job/qe-mdr-setup/750/testReport/ - It has all the logs.
I am afraid we will have leftovers as we are creating unique_resources to cut out the duplication when multiple apps are deployed on both the managed clusters. --> Created issue for this.